catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/monitor/email/SubscribedWorkspaceNotifierTest.java

```diff
@@ -0,0 +1,163 @@
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.Subject;
+import ddf.security.SubjectIdentity;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.shiro.util.ThreadContext;
+import org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceConstants;
+import org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceMetacardImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.opengis.filter.And;
+import org.opengis.filter.Filter;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SubscribedWorkspaceNotifierTest {
+
+  private SubscribedWorkspaceNotifier subscribedWorkspaceNotifier;
+
+  @Mock private EmailNotifier mockEmailNotifier;
+
+  @Mock private QueryResponse mockQueryResponse;
+
+  @Mock private CatalogFramework mockCatalogFramework;
+
+  @Mock private SubjectIdentity mockSubjectIdentity;
+
+  @Mock private QueryRequest mockQueryRequest;
+
+  @Mock private Attribute mockAttribute;
+
+  @Before
+  public void setup()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+    WorkspaceMetacardImpl mockWorkspaceMetacard = mock(WorkspaceMetacardImpl.class);
+    QueryResponse mockWorkspaceMetacardQueryResponse = mock(QueryResponse.class);
+    ExpressionBuilder mockExpressionBuilder = mock(ExpressionBuilder.class);
+    AttributeBuilder mockAttributeBuilder = mock(AttributeBuilder.class);
+    FilterBuilder mockFilterBuilder = mock(FilterBuilder.class);
+    Subject mockSubject = mock(Subject.class);
+    Result mockResult = mock(Result.class);
+
+    ThreadContext.bind(mockSubject);
+    List<Result> resultsForMockQueryResponse = Collections.singletonList(mockResult);
+    Map<String, Serializable> propertiesForMockQueryRequest = new HashMap<>();
+    propertiesForMockQueryRequest.put("id", "1234abcd");
+    List<Serializable> valuesForMockAttribute = Collections.singletonList("test@email.com");
+    Set<String> metacardTagsForMockWorkspaceMetacard =
+        Collections.singleton(WorkspaceConstants.WORKSPACE_TAG);
+
+    when(mockFilterBuilder.allOf(any(Filter.class), any(Filter.class))).thenReturn(mock(And.class));
+    when(mockFilterBuilder.attribute(anyString())).thenReturn(mockAttributeBuilder);
+    when(mockAttributeBuilder.is()).thenReturn(mockExpressionBuilder);
+    when(mockExpressionBuilder.text(anyString())).thenReturn(mock(Filter.class));
+    when(mockQueryResponse.getResults()).thenReturn(resultsForMockQueryResponse);
+    when(mockQueryResponse.getRequest()).thenReturn(mockQueryRequest);
+    when(mockQueryRequest.getProperties()).thenReturn(propertiesForMockQueryRequest);
```

Comment:
I think you can just use `Collections.singleton("id", "1234abcd")` here, then you can remove the map variable on line 90.